snuba/clusters/storage_sets.py

```diff
@@ -29,3 +30,25 @@ class StorageSetKey(Enum):
 
 # Storage sets enabled only when development features are enabled.
 DEV_STORAGE_SETS: FrozenSet[StorageSetKey] = frozenset({StorageSetKey.METRICS})
+
+# Storage sets in a group share the same query and distributed nodes but
+# do not have the same local node cluster configuration.
+# Joins can be performed across storage sets in the same group.
+STORAGE_SET_GROUPS: FrozenSet[Tuple[StorageSetKey, ...]] = frozenset(
+    {(StorageSetKey.EVENTS, StorageSetKey.EVENTS_RO, StorageSetKey.CDC)}
+)
+
+
+def is_valid_storage_set_combination(*storage_sets: StorageSetKey) -> bool:
+    first = storage_sets[0]
+
+    for storage_set in storage_sets[1:]:
+        if storage_set == first:
+            continue
+
+        for group in STORAGE_SET_GROUPS:
+            if first in group and storage_set in group:
+                continue
+            return False
+
+    return True
```

Comment:
Isn't this expressed in a simpler way as:
- build a sets with all the stroage sets in input (to avoid duplicates)
- scan once the STORAGE_SET_GROUPS and for each storage set group return true if it is a superset of the set in input.